File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>freq;
        vector<pair<int,int>>eandf;
        vector<int>output;

        for(int num : nums){
            freq[num]++;
        }

        eandf.insert(eandf.end(),freq.begin(),freq.end());

        std::sort(eandf.begin(), eandf.end(), [](const std::pair<int, int>& a, const std::pair<int, int>& b) {
            return a.second > b.second;
        }); 

        for(int i = 0;i<k;i++){
            output.push_back(eandf[i].first);
        }
    return output;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-0.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer,Integer> freq = new HashMap<>();
        List<Integer> res = new ArrayList<>();

        for(int num:nums){
            if(!freq.containsKey(num)){
                freq.put(num,1);
                continue;
            }
            freq.put(num,freq.get(num)+1);
        }

        for(int num:nums){
            if(freq.get(num)>nums.length/3){
                res.add(num);
                freq.put(num,0);
            }
        }

        return res;
    }
}